### ISSUE TITLE
First cached entry using GetOrSet should trigger Miss event

### ISSUE THREAD

**Issue by JoeShook** [jodydonetti/ZiggyCreatures.FusionCache#49]
The following code will trigger the SET event and I thing that is good and correct.  But it does not trigger the Miss event.  I think it should trigger the Miss event if it did not previously exist in the cache at anytime.  Think about it this way.  If you have a cache item in a cache it had to get there because once upon a time it as not in the cache, thus the rational that it should have historicaly been a cache miss.  I discovered this while building my OpenTelemetry plugin examples.  I am using the GetOrSet now that I have the Adaptive Caching feature in my hands.  That is when I realized couldn't find a Miss events in my prometheus (time series) database.  

````csharp
await cache.GetOrSetAsync<int>(
    "foo",
    async (_) =>
    {
        await Task.Delay(1);
        return 123;
    });
````

**Comment by jodydonetti** [comment]
Hi @JoeShook , I've just made the change on a local copy: will publish it in the next days I hope.
Is there something else that you discovered regarding events since you created this issue?
Thanks!

**Comment by jodydonetti** [comment]
> I checked the Main branch against the [Metrics Plugin](https://github.com/JoeShook/ZiggyCreatures.FusionCache.Metrics) tests in my develop branch. It looks good.
> 
> I didn't have another event issue at the moment.

That's great 🎉

> FYI, when I get some time I plan on having the [opentelemetry](https://github.com/JoeShook/ZiggyCreatures.FusionCache.Metrics/tree/otelWithExample) plugin live in [opentelemetry-dotnet-contrib](https://github.com/pengweiqhca/opentelemetry-dotnet-contrib). The metrics part of the OTel plugin was easy to produce. But I was wondering if I could hook "tracing" better. Maybe just adding a tag to an existing Activity so that things like HttpClient and/or EF traces would have a tag indicating it was FusionCache as the caller. Just need to find some time. Being in the contrib project would probably give FusionCache even more exposure.

Wow, that is awesome @JoeShook and it would be great if it will be included there!

**Comment by jodydonetti** [comment]
Hi @JoeShook , I just released [v0.11.0](https://github.com/jodydonetti/ZiggyCreatures.FusionCache/releases/tag/v0.11.0) which includes the fix for this issue 🎉

Hope this helps.
